modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/MultiSelectTag.java

```diff
@@ -0,0 +1,337 @@
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		String id = getId();
+
+		if (_label != null) {
+			jspWriter.write("<label");
+
+			if (Validator.isNotNull(id)) {
+				jspWriter.write(" for=\"");
+				jspWriter.write(id);
+				jspWriter.write("\"");
+			}
+
+			jspWriter.write(">");
+
+			jspWriter.write(_label);
+			jspWriter.write("</label>");
+		}
+
+		jspWriter.write("<div class=\"input-group\"><div class=\"");
+		jspWriter.write("input-group-item\"><div class=\"form-control ");
+		jspWriter.write("form-control-tag-group input-group\"><div class=\"");
+		jspWriter.write("input-group-item\">");
+
+		List<Map<String, String>> selectedItems = getSelectedItems();
+
+		if (!ListUtil.isEmpty(selectedItems)) {
+			for (Map<String, String> selectedItem : selectedItems) {
+				String selectedItemLabel = selectedItem.get("label");
+
+				LabelTag labelTag = new LabelTag();
+
+				labelTag.setDismissible(true);
+				labelTag.setLabel(selectedItemLabel);
+
+				labelTag.doTag(pageContext);
+
+				jspWriter.write("<input type=\"hidden\" name=\"");
+				jspWriter.write(selectedItemLabel);
+				jspWriter.write("\" ");
+				jspWriter.write("value=\"");
+				jspWriter.write(selectedItem.get("value"));
+				jspWriter.write("\" />");
+			}
+		}
+
+		jspWriter.write("<input class=\"form-control-inset\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(id)) {
+			super.writeIdAttribute();
+		}
+
+		jspWriter.write(" type=\"text\" /></div>");
+
+		if (!_disabled && !_disabledClearAll && !selectedItems.isEmpty()) {
+			jspWriter.write("<div class=\"input-group-item ");
+			jspWriter.write("input-group-item-shrink\">");
+
+			ButtonTag buttonTag = new ButtonTag();
+
+			buttonTag.setCssClass("component-action");
+			buttonTag.setDisplayType("");
+			buttonTag.setIcon("times-circle");
+
+			if (Validator.isNotNull(_clearAllTitle)) {
+				String clearAllTitle = LanguageUtil.get(
+					TagResourceBundleUtil.getResourceBundle(pageContext),
+					_clearAllTitle);
+
+				buttonTag.setTitle(HtmlUtil.escape(clearAllTitle));
+			}
+
+			buttonTag.doTag(pageContext);
+
+			jspWriter.write("</div>");
+		}
+
+		jspWriter.write("</div>");
+
+		if (_helpText != null) {
+			jspWriter.write("<div class=\"form-feedback-group\"><div ");
+			jspWriter.write("class=\"form-text\">");
+			jspWriter.write(_helpText);
+			jspWriter.write("</div></div>");
+		}
+
+		jspWriter.write("</div></div>");
+
+		return SKIP_BODY;
+	}
+
+	@Override
+	protected void writeDynamicAttributes() {
+	}
+
+	@Override
+	protected void writeIdAttribute() {
+	}
+
+	private static final String _ATTRIBUTE_NAMESPACE = "clay:multi-select:";
+
+	private String _clearAllTitle = "";
+	private String _closeButtonAriaLabel = "";
+	private boolean _disabled;
+	private boolean _disabledClearAll;
+	private String _helpText = "";
+	private String _inputName = "";
+	private String _inputValue = "";
+	private String _label = "";
+	private Object _locator;
```

Comment:
We should we create a `Locator` util class to be sure that any passed locator implements `label` and `value` properties (as is required for the clay component), defaults to `label` and `value` respectively.

Reply:
From what I understood, the `locator` is not going to be provide any value to us server-side. The only example we have on Clay about it is to be used to customize where the `label` and `value` are rendered in the DOM, but in SSR we only see the input with the `selectedItems`, no dropdown that will have the custom DOM.

Reply:
Not exactly. Locator is used to tell the component what are the properties that will provide the label and the value.

If you pass items (selectedItems or sourceItems) with `label` and `value` you don't need to provide any locator, but if you want those value to be get from other properties of the item (`name` and `value` for example) you need to pass a locator specifying `label: 'name', value: 'value'`. So you never get an item `label` the `value` from `selectedItem.label` or `sourceItem.label` but from `selectedItem[locator.label]` and so on...

Reply:
Let's see if I can do that correctly

Reply:
@carloslancha when you can please check [this commit](https://github.com/liferay-frontend/liferay-portal/pull/528/commits/1e4c07eee6bb66194d2494c4c63c54b94484708a) and tell me what you think.
I've also updated the "sample" so that one example uses this.